lib/logstash/filters/aggregate.rb

```diff
@@ -204,8 +204,68 @@
 # * The key point is that each time aggregate plugin detects a new `country_name`, it pushes previous aggregate map as a new Logstash event, and then creates a new empty map for the next country
 # * When 5s timeout comes, the last aggregate map is pushed as a new event
 # * Finally, initial events (which are not aggregated) are dropped because useless (thanks to `event.cancel()`)
-# 
-# 
+#
+#
+# ==== Example #5 : no end event and push events as soon as possible
+#
+# Fifth use case: like example #3, there is no end event. Events keep comming for an idefinite time and you want to push the aggregation map as soon as possbile after the last user interaction without waiting for the `timeout`. This allows to have the aggregated events pushed closer to real time.
+#
+# A typical case is aggregating or tracking user behaviour. We can track a user by its ID through the events, however once the user stops interacting, the events stop coming in. There is no specific event indicating the end of the user's interaction. The user ineraction will be considered as ended when no events for the specified user (task_id) arrive afer the specified inactivity_timeout`.
+#
+# If the user keeps interacting for longer than `timeout` the aggration map will be pushed at timoeout.
```

Comment:
Replace the whole line by :
If the user continues interacting for longer than `timeout` seconds (since task creation time), the aggregation map will still be deleted and pushed as a new event when timeout occurs.